omnipaxos_core/src/sequence_paxos/follower.rs

```diff
@@ -148,16 +148,11 @@ where
         if self.internal_storage.get_promise() == acc.n
             && self.state == (Role::Follower, Phase::Accept)
         {
-            if acc.seq_num == ACCSYNC_SEQ_NUM {
+            if acc.seq_num.counter == PREDEFINED_LEADER_FIRST_ACCEPT {
                 // psuedo-AcceptSync for reconfigurations
-                #[cfg(feature = "logging")]
-                debug!(
-                    self.logger,
-                    "Incoming message first Accept of reconfiguration"
-                );
                 self.internal_storage.set_accepted_round(acc.n);
                 self.forward_pending_proposals();
-            } else if self.current_seq_num + 1 != acc.seq_num {
+            } else if !acc.seq_num.is_successor_of(self.current_seq_num) {
```

Comment:
To avoid doing `reconnected()` due to old messages with lower session or seq numbers I suggest we have a function:

```
enum MessageStatus {
      Expected,   // happy case
      DroppedPreceding,    // do reconnected
      Outdated,   // lower SequenceNumber can be ignored
}

fn check_msg_status(&mut self, msg_seq_num: SequenceNumber) -> MessageStatus 
```